.github/workflows/tests.yml

```diff
@@ -9,32 +9,33 @@ on:
 env:
   PY_COLORS: "1"
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   tests:
     name: tests
     runs-on: "ubuntu-latest"
     defaults:
       run:
-        shell: bash -l {0}
+        shell: bash -leo pipefail {0}
 
     steps:
-      - name: cancel previous runs
-        uses: styfle/cancel-workflow-action@0.6.0
-        with:
-          access_token: ${{ github.token }}
-
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
-      - uses: conda-incubator/setup-miniconda@v3
+      - uses: mamba-org/setup-micromamba@v1
         with:
-          miniforge-version: latest
-          environment-file: environment.yml
-          python-version: "${{ vars.PYVER }}"
-          activate-environment: cf-scripts
+          # environment-file: environment.yml
+          environment-name: cf-scripts
           condarc-file: autotick-bot/condarc
+          create-args: >-
+            conda
+            conda-lock
 
```

Comment:
With locking, I always have all dependencies in the environment.yml as otherwise further install steps lead to spurious updates again.